onnx/defs/text/defs.cc

```diff
@@ -33,6 +33,69 @@ ONNX_OPERATOR_SET_SCHEMA(
                 *ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape());
         }));
 
+static const char* StringSplit_doc =
+    R"DOC(StringSplit splits a string tensor's elements into substrings based on a delimiter attribute and a maxsplit attribute. The first output of this operator is a tensor of strings representing the substrings from splitting each input string on the delimiter substring. A integer tensor is also returned representing the number of substrings generated. This output tensor has one additional rank compared to the input to store these substrings, as examples below will illustrate.)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    StringSplit,
+    20,
+    OpSchema()
+        .Input(0, "X", "Tensor of strings to split.", "T1", OpSchema::Single, true, 1, OpSchema::NonDifferentiable)
+        .Attr(
+            "delimiter",
+            "Delimiter to split on. If left unset this defaults to a space character.",
+            AttributeProto::STRING,
+            false)
+        .Attr(
+            "maxsplit",
+            "Maximum number of splits (from left to right). If left unset, it will make as many splits as possible.",
```

Comment:
If this attribute is specified, is the (last) output-dimension guaranteed to be "maxsplit" (even if no input string has that many split substrings)? Please clarify in the description. Further, if that is the case, the shape-inference can produce the more precise output-shape when this attribute is specified.

Reply:
No, it merely specifies the maximum number of splits. I'll update the description to clarify this.

e.g. `StringSplit(["a", "b"], maxsplit=2) => [["a"], ["b"]], [1, 1]`